packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -0,0 +1,220 @@
+  readonly hostname: string;
+  /**
+   * The service discovery namespace name, this is used for service discovery
+   *
+   * @example domain.local
+   */
+  readonly namespaceName: string;
+  /**
+   * Array of VirtualNodeBackendProps
+   */
+  readonly backends?: VirtualNodeBackendProps[];
+  /**
+   * Listener properties, such as portMappings and optional healthChecks
+   */
+  readonly listeners?: ListenerProps;
+}
+
+/**
+ * VritualNode represents a newly createded AppMesh VirtualNode
+ *
+ * @see https://docs.aws.amazon.com/app-mesh/latest/userguide/virtual_nodes.html
+ */
+export class VirtualNode extends cdk.Construct {
+  /**
+   * The name of the AppMesh which the virtual node belongs to
+   *
+   * @type {string}
+   * @memberof VirtualNode
+   */
+  public readonly meshName: string;
+  /**
+   * The name of the VirtualNode
+   *
+   * @type {string}
+   * @memberof VirtualNode
+   */
+  public readonly virtualNodeName: string;
+  /**
+   * The Amazon Resource Name belonging to the VirtualNdoe
+   *
+   * @type {string}
+   * @memberof VirtualNode
+   */
+  public readonly virtualNodeArn: string;
+
+  private readonly backends: CfnVirtualNode.BackendProperty[] = [];
+  private readonly listeners: CfnVirtualNode.ListenerProperty[] = [];
+  private readonly namespaceName: string;
+
+  constructor(scope: cdk.Construct, id: string, props: VirtualNodeProps) {
+    super(scope, id);
+
+    this.meshName = props.meshName;
+    this.namespaceName = props.namespaceName;
+
+    if (props.backends) {
+      this.addBackends(props.backends);
+    }
+
+    if (props.listeners) {
+      this.addListeners(props.listeners);
+    }
+
+    this.node.apply(new cdk.Tag(NAME_TAG, this.node.path));
+    const name = props.nodeName ? props.nodeName : id;
+
+    const node = new CfnVirtualNode(this, 'VirtualNode', {
+      virtualNodeName: name,
+      meshName: this.meshName,
+      spec: {
+        backends: this.backends,
+        listeners: this.listeners,
+        serviceDiscovery: {
+          dns: {
+            hostname: `${props.hostname}.${this.namespaceName}`,
+          },
+        },
+        logging: {
+          accessLog: {
+            file: {
+              path: '/dev/stdoout',
+            },
+          },
+        },
+      },
+    });
+
+    this.virtualNodeName = node.virtualNodeName;
+    this.virtualNodeArn = node.virtualNodeArn;
+  }
+
+  /**
+   * Utility method to add a single backend for existing or new VritualNodes
+   *
+   * @param {VirtualNodeBackendProps} props
+   * @memberof VirtualNode
+   */
+  public addBackend(props: VirtualNodeBackendProps) {
+    this.backends.push({
+      virtualService: {
+        virtualServiceName: props.virtualServiceName,
+      },
+    });
+  }
+
+  /**
+   * Utility method to add backends for existing or new VritualNodes
+   *
+   * @param {VirtualNodeBackendProps[]} props
+   * @memberof VirtualNode
+   */
+  public addBackends(props: VirtualNodeBackendProps[]) {
+    props.forEach(s => {
```

Comment:
Please use `for (s of props)` instead of `props.forEach`.

Reply:
I'll do the `for of` no problem, though just for my understand and own edification if you don't mind.

Any specific reasoning? readability? performance? I'll change, either way, your team owns the code here, I'm really just curious.